### PR TITLE
Add ESLint support - #404

### LINKED ISSUE #404: Add eslint support
I really like facebook's graphql eslint rules and we use them in couple of projetcs.
See: https://github.com/graphql/graphql-js/blob/master/.eslintrc

But, I'm open to some other set of rules as well. Meteor also seems to  be using their own set of rules as proposed(or will  be) on the [Meteor Guide](https://github.com/meteor/todos).  We can use that as well.

I heard, in Meteor there is a built in linting tooling in the build API. May be we can use that as well.

### PR DESCRIPTION
**Pull request by johanbrook**
Fixes #404.

This patch adds ESLint support for Flow Router, which means:

- ESLint config
   - Rules from [the Meteor Guide example app](https://github.com/meteor/todos/blob/master/.eslintrc-meteor). Heavily angled against ES6.
   - [ESLint plugin for Meteor](https://github.com/dferber90/eslint-plugin-meteor) for Meteor specific analysis.
- a `npm run lint` command
- a `npm test` command, just for funsies.

There are ~1400 linting warnings and errors at this time of writing, so I reckon these will be fixed during time. Also, we can modify the rules to fit this project as well.

This patch adds some npm developer dependencies, so in order to get the linting stuff, please run

```
npm install
```

after merge.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{
+  "name": "flow-router",
+  "description": "Carefully Designed Client Side Router for Meteor.",
+  "scripts": {
+    "test": "spacejam test-packages ./ --port 9001",
```
Comment (arunoda) [inline package.json]: Mark this as `testonly`.
Add a new script with `test` which does both `lint` and `testonly`

**Comment by arunoda** [comment]
You can do `npm lint -- --fix` and fix most of the common issues. Submit with them too.
I'll work on other errors.

**Comment by arunoda** [comment]
Once you done these changes, add a comment. Otherwise, GH does not send any notification.

**Comment by arunoda** [comment]
I took your commit manually and applied with lint fixes.
See: https://github.com/kadirahq/flow-router/pull/436

Thanks.

**Comment by johanbrook** [comment]
Hi @arunoda, sorry for being late on the follow-up.

I take there's no need merging this to `master` now?

**Comment by arunoda** [comment]
@johanbrook no problem. Yeah. We started development on the `ssr` and @elie222 converted it to use ES2015. 

But tests are not yet written in ES2015. So, I ignored that from the linting.

**Pull request closed without merging**